salt/requisite.py

```diff
@@ -0,0 +1,412 @@
+                RequisiteType.PREREQ,
+                RequisiteType.ONFAIL,
+                RequisiteType.ONCHANGES_ANY,
+                RequisiteType.ONFAIL_ALL,
+                RequisiteType.ONCHANGES,
+                RequisiteType.ONCHANGES_ANY)
+        }
+        for r_type in reqs:
+            if low_reqs := low.get(r_type.value):
+                if r_type in disabled_reqs:
+                    log.warning(
+                        "The %s requisite has been disabled, Ignoring.", r_type
+                    )
+                    continue
+                for req_ref in low_reqs:
+                    if isinstance(req_ref, str):
+                        req_ref = {"id": req_ref}
+                    req_ref = trim_req(req_ref)
+                    # req_key: match state module name
+                    # req_val: match state id or name
+                    req_key, req_val = next(iter(req_ref.items()))
+                    if req_val is None:
+                        continue
+                    if not isinstance(req_val, str):
+                        return (
+                            f"Requisite [{r_type}: {req_key}] in state"
+                            f" [{low['name']}] in SLS [{low.get('__sls__')}]"
+                            " must have a string as the value"
+                        )
+                    found = self.add_dependency(
+                        low, r_type, req_key, req_val)
+                    if not found:
+                        return ("Referenced state does not exist"
+                                f" for requisite [{r_type}: ({req_key}: {req_val})] in state"
+                                f" [{low['name']}] in SLS [{low.get('__sls__')}]")
+        return None
+
+    def aggregate_and_order_chunks(self, cap: int) -> list[LowChunk]:
+        """
+        Aggregate eligible nodes in the dependencies graph.
+
+        Return a list of the chunks in the sorted order in which the
+        chunks should be executed.
+        Nodes are eligible for aggregation if the state function in the
+        chunks match and aggregation is enabled in the configuration for
+        the state function.
+        :param cap: the maximum order value configured in the states
+        :return: the ordered chunks
+        """
+        dag: nx.MultiDiGraph = self.dag
+        # dict for tracking topo order and for mapping each node that
+        # was aggregated to the aggregated node that replaces it
+        topo_order = {}
+
+        max_group_size = 500
+        groups_by_type = defaultdict(list)
+
+        def _get_order(node):
+            chunk = dag.nodes[node].get("chunk", {})
+            chunk_order = self._get_chunk_order(cap, node)
+            return (chunk_order, "{0[state]}{0[name]}{0[fun]}".format(chunk))
+
+        # Iterate over the nodes in topological order to get the correct
+        # ordering which takes requisites into account
+        for i, node in enumerate(
```

Comment:
Why enumerate if you're not using `i`?

Reply:
This went through several iterations and at one point I believe it was using the `i`. It looks like it can be cleaned up and removed now.